src/west/manifest-schema.yml

```diff
@@ -130,3 +130,31 @@ mapping:
           clone-depth:
             required: false
             type: int
+
+  # The "self" key specifies values for the project containing the manifest
+  # file.
+  #
+  # When using West to initialize a project there must be a repository
+  # containing a manifest files with sub-projects.
+  # In case the Git repository containing the manifest file must be located in
+  # The path field can be used to specify a local path, in which the repository
+  # containing the manifest file must be placed.
+  #
+  # Example:
+  #
+  #   self:
+  #     - path: project-path
+  #
+  # Note: if "self" key is not present, the repository will be placed in a local
+  # folder with a name based upon the remote repository provided with
+  # `west init`, example:
+  # `west init -m https://example.com/project-repo` will clone the remote
+  # repository into the local path `<project>/project-repo`
+  #
+  self:
+    required: false
+    type: map
+    mapping:
+      path:
+        required: true
```

Comment:
I'm not sure making path required is a good idea. Right now it is the only key in the self map's value, sure, but later on we might want to add a second one. If we do that, users will still have to specify path, even if they want the default path.

Reply:
I also think we discussed this, and agreed it was not optional, but can't remember the details.
Although, I kind of like that `path` (and maybe even `self` )is mandatory, as in zephyr, we require a folder named `zephyr` containing `zephyr`.
And if user's fork `zephyr` to `zephyr-my-fork` and `self` is not present, then zephyr-base will fail.

Reply:
Makes no sense to have `self` be optional and `path` required. I would go for `path` being optional at this stage, defaulting to the repo name

Reply:
`path` is now optional, although if `self:` is specified, then `path` is also needed as there are no other keys specified for the map, schema error if one does:
```
self: 
```
without specifying anything.
So in practice `path` is required if specifying `self`

Reply:
Having self without path is strange, for sure, but should it really be an error?

For analogy, a manifest with `defaults:` but nothing underneath it is still not an error.

Reply:
This happens automatically by pykwalify.
I have nothing to do with it, just wanted to inform you ;)

` - Value 'None' is not a dict. Value path: '/self'.: Path: '/'>`